component/azstorage/block_blob.go

```diff
@@ -85,11 +85,16 @@ func (bb *BlockBlob) Configure(cfg AzStorageConfig) error {
 	bb.Config = cfg
 
 	bb.blobAccCond = azblob.BlobAccessConditions{}
-	bb.blobCPKOpt = azblob.ClientProvidedKeyOptions{}
+	bb.blobCPKOpt = azblob.ClientProvidedKeyOptions{
```

Comment:
This shall be set only if cpk is enabled thorugh config or cli.

Reply:
Yes cpk is supported either via config or cli. Is there anything else needed here ?

Reply:
What I meant is there shall be check that if cpk flag is enable then set it with these options otherwise continue using the empty data structure.